### PR TITLE
[#105] Optimized preprocessing;

### PR DESCRIPTION
**Pull request by Samuel-Campbell**
- Optimized time complexity
- Implemented standardization of words by Arek
- Facts and Precedence now stored in a dictionary to easily convert to jason
- Facts can only appear once

### CODE REVIEW

File: src/ml_service/WordVectors/FrenchVectors.py [hunk 1]
```diff
@@ -37,7 +39,7 @@ def get_stop_words():
             'pourquoi', 'quand', 'que', 'quel', 'quelle,''quelles', 'quels',
             'qui', 'sa', 'sans', 'ses', 'seulement', 'si', 'sien,''son', 'sont',
```
Comment (TaimoorRana) [inline src/ml_service/WordVectors/FrenchVectors.py]: extra `' `beside "son", "quelles'

File: src/ml_service/feature_extraction/Clustering/hdbscan/Hdbscan.py [hunk 2]
```diff
@@ -145,8 +145,8 @@ def plot(self, matrix, sent):
 
     start = time.time()
 
-    tpl = parser.parse_topics(Global.Precedence_Directory, number_of_files)
-    clusterer.train(cluster_dir, tpl, number_of_files, config)
+    #tpl = parser.parse_files(Global.Precedence_Directory, number_of_files)
```
Comment (TaimoorRana) [inline src/ml_service/feature_extraction/Clustering/hdbscan/Hdbscan.py]: please remove commented code

File: src/ml_service/feature_extraction/Preprocessing/Sam_Parser/PrecedenceParse.py [hunk 3]
```diff
@@ -157,28 +150,24 @@ def __split_sub_sentence(self, sentence):
...
-                data.append(vec)
-                sent.append(model.topics[i])
-                original_sent.append(model.original_topic[i])
-        return np.array(data), np.array(sent), np.array(original_sent)
+            self.__parse(i)
+        return self.__model
+
+
+if __name__ == '__main__':
+    p = Precedence_Parser()
```
Comment (TaimoorRana) [inline src/ml_service/feature_extraction/Preprocessing/Sam_Parser/PrecedenceParse.py]: please use more descriptive names for p and m.

**Review by TaimoorRana — APPROVED** [review]

**Review by arekmano — APPROVED** [review]
Looks good

**Comment by lancelafontaine** [comment]
Merging because npm package checksums are out-of-date and failing our build.

**Pull request merged**